[33mWARNING [0m juju.client.connection:connection.py:858 unexpected facade SSHServer received from the controller
[32mINFO    [0m pytest_operator.plugin:plugin.py:795 Connecting to existing model concierge-lxd:testing on unspecified cloud
[33mWARNING [0m juju.client.connection:connection.py:858 unexpected facade SSHServer received from the controller
[33mWARNING [0m juju.client.connection:connection.py:858 unexpected facade SSHSession received from the controller
[33mWARNING [0m juju.client.connection:connection.py:858 unexpected facade SSHTunneler received from the controller
[33mWARNING [0m juju.client.connection:connection.py:858 unexpected facade SSHServer received from the controller
[32mINFO    [0m juju.model:__init__.py:2301 Deploying ch:amd64/jammy/postgresql-764
[32mINFO    [0m juju.model:__init__.py:2301 Deploying local:jammy/pgbouncer-0
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  postgresql/0 [allocating] waiting: waiting for machine
  postgresql/1 [allocating] waiting: waiting for machine
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  postgresql/0 [allocating] waiting: waiting for machine
  postgresql/1 [allocating] waiting: waiting for machine
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  postgresql/0 [allocating] waiting: waiting for machine
  postgresql/1 [allocating] waiting: waiting for machine
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  postgresql/0 [allocating] waiting: waiting for machine
  postgresql/1 [executing] maintenance: installing PostgreSQL
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  postgresql/0 [allocating] waiting: waiting for machine
  postgresql/1 [executing] active: Primary (degraded)
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  postgresql/0 [executing] maintenance: installing PostgreSQL
  postgresql/1 [executing] active: Primary (degraded)
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  postgresql/0 [executing] waiting: waiting to start PostgreSQL
  postgresql/1 [executing] maintenance: reconfiguring cluster
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  postgresql/0 [executing] waiting: waiting to start PostgreSQL
  postgresql/1 [executing] maintenance: reconfiguring cluster
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  postgresql/0 [executing] waiting: waiting to start PostgreSQL
  postgresql/1 [executing] maintenance: reconfiguring cluster
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  postgresql/0 [executing] waiting: waiting to start PostgreSQL
  postgresql/1 [idle] active: Primary (degraded)
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  postgresql/0 [executing] waiting: waiting to start PostgreSQL
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  postgresql/0 [executing] waiting: awaiting for member to start
  postgresql/1 [idle] active: Primary
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  postgresql/0 [executing] waiting: waiting for primary to be reachable from this unit
  postgresql/1 [executing] active: 
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  postgresql/1 [idle] active: Primary
[33mWARNING [0m juju.model:__init__.py:1683 relate is deprecated and will be removed. Use integrate instead.
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  postgresql/0 [idle] active: 
  postgresql/1 [idle] active: Primary
[33mWARNING [0m juju.model:__init__.py:1683 relate is deprecated and will be removed. Use integrate instead.
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  landscape-server/0 [allocating] waiting: waiting for machine
  landscape-server/1 [allocating] waiting: waiting for machine
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  postgresql/0 [idle] active: 
  postgresql/1 [idle] active: Primary
  pgbouncer (not enough units yet - 0/1)
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  pgbouncer (not enough units yet - 0/1)
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  landscape-server/0 [allocating] waiting: waiting for machine
  landscape-server/1 [allocating] waiting: waiting for machine
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  landscape-server/0 [allocating] waiting: waiting for machine
  landscape-server/1 [allocating] waiting: waiting for machine
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  pgbouncer (not enough units yet - 0/1)
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  landscape-server/0 [allocating] waiting: waiting for machine
  landscape-server/1 [allocating] waiting: waiting for machine
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  pgbouncer (not enough units yet - 0/1)
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  landscape-server/0 [allocating] waiting: waiting for machine
  landscape-server/1 [allocating] waiting: waiting for machine
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  pgbouncer (not enough units yet - 0/1)
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  landscape-server/0 [allocating] waiting: waiting for machine
  landscape-server/1 [allocating] waiting: waiting for machine
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  pgbouncer (not enough units yet - 0/1)
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  landscape-server/0 [executing] maintenance: Installing apt packages
  landscape-server/1 [allocating] waiting: waiting for machine
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  pgbouncer (not enough units yet - 0/1)
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  landscape-server/0 [executing] maintenance: Installing apt packages
  landscape-server/1 [executing] maintenance: Installing apt packages
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  pgbouncer (not enough units yet - 0/1)
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  landscape-server/0 [executing] maintenance: Installing apt packages
  landscape-server/1 [executing] maintenance: Installing apt packages
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  pgbouncer (not enough units yet - 0/1)
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  landscape-server/0 [executing] maintenance: Installing apt packages
  landscape-server/1 [executing] maintenance: Installing apt packages
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  pgbouncer (not enough units yet - 0/1)
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  landscape-server/0 [executing] maintenance: Installing apt packages
  landscape-server/1 [executing] maintenance: Installing apt packages
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  pgbouncer (not enough units yet - 0/1)
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  landscape-server/0 [executing] waiting: Waiting on relations: db, amqp
  landscape-server/1 [executing] maintenance: Installing apt packages
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  pgbouncer/0 [executing] maintenance: Installing and configuring PgBouncer
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  landscape-server/0 [executing] waiting: Waiting on relations: db, amqp
  landscape-server/1 [executing] waiting: Waiting on relations: db, amqp
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  postgresql/0 [executing] active: 
  postgresql/1 [executing] active: Primary
  pgbouncer/0 [executing] blocked: waiting for backend database relation to initialise
  pgbouncer/1 [executing] maintenance: Installing and configuring PgBouncer
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  landscape-server/0 [idle] waiting: 
  landscape-server/1 [executing] waiting: Waiting on relations: db
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  postgresql/0 [executing] active: 
  postgresql/1 [executing] active: Primary
  pgbouncer/0 [idle] active: 
  pgbouncer/1 [executing] maintenance: Reloading Pgbouncer
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  landscape-server/0 [executing] maintenance: Setting up databases
  landscape-server/1 [executing] maintenance: Setting up databases
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  postgresql/0 [executing] active: 
  postgresql/1 [executing] active: Primary
  pgbouncer/0 [executing] active: 
  pgbouncer/1 [executing] active: 
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  landscape-server/0 [executing] maintenance: Setting up databases
  landscape-server/1 [executing] maintenance: Setting up databases
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  postgresql/0 [executing] active: 
  postgresql/1 [idle] active: Primary
  pgbouncer/0 [executing] active: 
  pgbouncer/1 [executing] active: 
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  landscape-server/0 [executing] maintenance: Setting up databases
  landscape-server/1 [executing] maintenance: Setting up databases
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  postgresql/0 [idle] active: 
  postgresql/1 [idle] active: Primary
  pgbouncer/0 [executing] active: 
  pgbouncer/1 [executing] active: 
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  landscape-server/0 [executing] maintenance: Setting up databases
  landscape-server/1 [executing] maintenance: Setting up databases
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  postgresql/0 [idle] waiting: awaiting for member to start
  postgresql/1 [executing] active: Primary (degraded)
  pgbouncer/0 [executing] active: 
  pgbouncer/1 [executing] active: 
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  landscape-server/0 [executing] maintenance: Starting services
  landscape-server/1 [executing] maintenance: Setting up databases
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  postgresql/0 [executing] active: 
  postgresql/1 [idle] active: 
  pgbouncer/0 [executing] active: 
  pgbouncer/1 [executing] active: 
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  landscape-server/0 [executing] maintenance: Starting services
  landscape-server/1 [executing] maintenance: Setting up databases
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  postgresql/1 [idle] active: 
  pgbouncer/0 [executing] active: 
  pgbouncer/1 [executing] active: 
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  landscape-server/0 [executing] maintenance: Starting services
  landscape-server/1 [executing] maintenance: Setting up databases
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  pgbouncer/0 [executing] active: 
  pgbouncer/1 [executing] active: 
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  landscape-server/0 [executing] maintenance: Starting services
  landscape-server/1 [executing] maintenance: Setting up databases
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  pgbouncer/0 [executing] active: 
  pgbouncer/1 [executing] active: 
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  landscape-server/0 [idle] active: Unit is ready
  landscape-server/1 [executing] maintenance: Starting services
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  pgbouncer/0 [idle] active: 
  pgbouncer/1 [executing] active: 
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  landscape-server/0 [idle] active: Unit is ready
  landscape-server/1 [executing] maintenance: Starting services
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  pgbouncer/1 [executing] active: 
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  landscape-server/0 [idle] active: Unit is ready
  landscape-server/1 [idle] active: Unit is ready
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  postgresql/1 [idle] active: Primary
  pgbouncer/0 [idle] active: 
  pgbouncer/1 [executing] active: 
[32mINFO    [0m root:helpers.py:96 (0, '{"pgbouncer/0":{"workload-version":"1.21.0","machine":"3","opened-ports":null,"public-address":"10.96.199.61","charm":"local:jammy/pgbouncer-0","leader":true,"life":"alive","relation-info":[{"relation-id":5,"endpoint":"backend-database","related-endpoint":"database","application-data":{"data":"{\\"database\\": \\"pgbouncer\\", \\"extra-user-roles\\": \\"SUPERUSER\\", \\"provided-secrets\\": \\"[\\\\\\"mtls-cert\\\\\\"]\\", \\"requested-secrets\\": \\"[\\\\\\"username\\\\\\", \\\\\\"password\\\\\\", \\\\\\"tls\\\\\\", \\\\\\"tls-ca\\\\\\", \\\\\\"uris\\\\\\", \\\\\\"read-only-uris\\\\\\"]\\"}","database":"pgbouncer","endpoints":"10.96.199.166:5432","read-only-endpoints":"10.96.199.88:5432","secret-tls":"secret://d8a7ddd7-a8d8-4c92-8b9b-074ff2897b45/d0tr6lc8qh2s43ac1en0","secret-user":"secret://d8a7ddd7-a8d8-4c92-8b9b-074ff2897b45/d0tr6l48qh2s43ac1emg","version":"14.17"},"local-unit":{"in-scope":false,"data":null},"related-units":{"postgresql/0":{"in-scope":true,"data":{"egress-subnets":"10.96.199.88/32","ingress-address":"10.96.199.88","private-address":"10.96.199.88"}},"postgresql/1":{"in-scope":true,"data":{"egress-subnets":"10.96.199.166/32","ingress-address":"10.96.199.166","private-address":"10.96.199.166"}}}},{"relation-id":11,"endpoint":"db-admin","related-endpoint":"db","application-data":{},"local-unit":{"in-scope":false,"data":null},"related-units":{"landscape-server/0":{"in-scope":true,"data":{"egress-subnets":"10.96.199.61/32","ingress-address":"10.96.199.61","private-address":"10.96.199.61"}},"landscape-server/1":{"in-scope":true,"data":{"egress-subnets":"10.96.199.116/32","ingress-address":"10.96.199.116","private-address":"10.96.199.116"}}}},{"relation-id":4,"endpoint":"pgb-peers","related-endpoint":"pgb-peers","application-data":{"current_port":"6432","leader_ip":"10.96.199.61","pgb_dbs_config":"{\\"11\\": {\\"name\\": \\"landscape-server\\", \\"legacy\\": true}, \\"*\\": {\\"name\\": \\"*\\", \\"auth_dbname\\": \\"landscape-server\\"}}","pgbouncer_user_11_testing":"{\\"user\\": \\"pgbouncer_user_11_testing\\", \\"password\\": \\"cPIzWGJcN3n3Q963nY1MZHPa\\", \\"database\\": \\"landscape-server\\"}","readonly_dbs":"[\\"landscape-standalone-account-1\\", \\"landscape-standalone-knowledge\\", \\"landscape-standalone-main\\", \\"landscape-standalone-package\\", \\"landscape-standalone-resource-1\\", \\"landscape-standalone-session\\"]"},"local-unit":{"in-scope":true,"data":{"auth_file_set":"true","egress-subnets":"10.96.199.61/32","ingress-address":"10.96.199.61","pgb_dbs":"7f1f4f98c564f94542b2b5c8199195bf","private-address":"10.96.199.61","userlist_nonce":"1gb9FKBGAH4uei8E7uOjmOsP"}},"related-units":{"pgbouncer/1":{"in-scope":true,"data":{"auth_file_set":"true","egress-subnets":"10.96.199.116/32","ingress-address":"10.96.199.116","pgb_dbs":"7f1f4f98c564f94542b2b5c8199195bf","private-address":"10.96.199.116","userlist_nonce":"vpUDis4b4Qs0B5CcIpHfp5qi"}}}},{"relation-id":3,"endpoint":"upgrade","related-endpoint":"upgrade","application-data":{"dependencies":"{\\"charm\\": {\\"dependencies\\": {}, \\"name\\": \\"pgbouncer\\", \\"upgrade_supported\\": \\"\\u003e0\\", \\"version\\": \\"1\\"}, \\"snap\\": {\\"dependencies\\": {}, \\"name\\": \\"charmed-postgresql\\", \\"upgrade_supported\\": \\"^1\\", \\"version\\": \\"1.18.0\\"}}"},"local-unit":{"in-scope":true,"data":{"egress-subnets":"10.96.199.61/32","ingress-address":"10.96.199.61","private-address":"10.96.199.61","state":"idle"}},"related-units":{"pgbouncer/1":{"in-scope":true,"data":{"egress-subnets":"10.96.199.116/32","ingress-address":"10.96.199.116","private-address":"10.96.199.116","state":"idle"}}}}]}}\n', '')
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  landscape-server/0 [executing] active: Unit is ready
  landscape-server/1 [executing] active: Unit is ready
  postgresql/0 [idle] active: 
  postgresql/1 [idle] active: Primary
  pgbouncer/0 [idle] active: 
  pgbouncer/1 [idle] active: 
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  landscape-server/0 [idle] active: Unit is ready
  landscape-server/1 [executing] maintenance: Starting services
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  landscape-server/0 [idle] active: Unit is ready
  landscape-server/1 [executing] maintenance: Starting services
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  landscape-server/0 [idle] active: Unit is ready
  landscape-server/1 [executing] maintenance: Starting services
[32mINFO    [0m juju.model:__init__.py:3254 Waiting for model:
  landscape-server/0 [idle] active: Unit is ready
  landscape-server/1 [executing] maintenance: Starting services
[32mINFO    [0m pytest_operator.plugin:plugin.py:937 Model status:

Model    Controller     Cloud/Region         Version  SLA          Timestamp
testing  concierge-lxd  localhost/localhost  3.6.6    unsupported  02:05:49Z

App               Version  Status  Scale  Charm             Channel        Rev  Exposed  Message
haproxy                    active      1  haproxy           latest/stable   75  yes      Unit is ready
landscape-server           active      2  landscape-server  latest/stable  124  no       Unit is ready
pgbouncer         1.21.0   active      2  pgbouncer                          0  no       
postgresql        14.17    active      2  postgresql        14/edge        764  no       
rabbitmq-server   3.9.27   active      1  rabbitmq-server   3.9/stable     188  no       Unit is ready

Unit                 Workload  Agent  Machine  Public address  Ports           Message
haproxy/0*           active    idle   2        10.96.199.100   80,443/tcp      Unit is ready
landscape-server/0*  active    idle   3        10.96.199.61                    Unit is ready
  pgbouncer/0*       active    idle            10.96.199.61                    
landscape-server/1   active    idle   4        10.96.199.116                   Unit is ready
  pgbouncer/1        active    idle            10.96.199.116                   
postgresql/0         active    idle   0        10.96.199.88    5432/tcp        
postgresql/1*        active    idle   1        10.96.199.166   5432/tcp        Primary
rabbitmq-server/0*   active    idle   5        10.96.199.4     5672,15672/tcp  Unit is ready

Machine  State    Address        Inst id        Base          AZ  Message
0        started  10.96.199.88   juju-897b45-0  ubuntu@22.04      Running
1        started  10.96.199.166  juju-897b45-1  ubuntu@22.04      Running
2        started  10.96.199.100  juju-897b45-2  ubuntu@22.04      Running
3        started  10.96.199.61   juju-897b45-3  ubuntu@22.04      Running
4        started  10.96.199.116  juju-897b45-4  ubuntu@22.04      Running
5        started  10.96.199.4    juju-897b45-5  ubuntu@22.04      Running

[32mINFO    [0m pytest_operator.plugin:plugin.py:943 Juju error logs:

machine-1: 01:44:45 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-1: 01:44:45 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-postgresql-1: 01:44:45 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
unit-postgresql-1: 01:45:25 ERROR unit.postgresql/1.juju-log Unable to get the state of the cluster
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-1/charm/venv/lib/python3.10/site-packages/urllib3/connection.py", line 198, in _new_conn
    sock = connection.create_connection(
  File "/var/lib/juju/agents/unit-postgresql-1/charm/venv/lib/python3.10/site-packages/urllib3/util/connection.py", line 85, in create_connection
    raise err
  File "/var/lib/juju/agents/unit-postgresql-1/charm/venv/lib/python3.10/site-packages/urllib3/util/connection.py", line 73, in create_connection
    sock.connect(sa)
ConnectionRefusedError: [Errno 111] Connection refused

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-1/charm/venv/lib/python3.10/site-packages/urllib3/connectionpool.py", line 787, in urlopen
    response = self._make_request(
  File "/var/lib/juju/agents/unit-postgresql-1/charm/venv/lib/python3.10/site-packages/urllib3/connectionpool.py", line 493, in _make_request
    conn.request(
  File "/var/lib/juju/agents/unit-postgresql-1/charm/venv/lib/python3.10/site-packages/urllib3/connection.py", line 445, in request
    self.endheaders()
  File "/usr/lib/python3.10/http/client.py", line 1278, in endheaders
    self._send_output(message_body, encode_chunked=encode_chunked)
  File "/usr/lib/python3.10/http/client.py", line 1038, in _send_output
    self.send(msg)
  File "/usr/lib/python3.10/http/client.py", line 976, in send
    self.connect()
  File "/var/lib/juju/agents/unit-postgresql-1/charm/venv/lib/python3.10/site-packages/urllib3/connection.py", line 276, in connect
    self.sock = self._new_conn()
  File "/var/lib/juju/agents/unit-postgresql-1/charm/venv/lib/python3.10/site-packages/urllib3/connection.py", line 213, in _new_conn
    raise NewConnectionError(
urllib3.exceptions.NewConnectionError: <urllib3.connection.HTTPConnection object at 0x7fd9b5e0ee90>: Failed to establish a new connection: [Errno 111] Connection refused

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-1/charm/venv/lib/python3.10/site-packages/requests/adapters.py", line 667, in send
    resp = conn.urlopen(
  File "/var/lib/juju/agents/unit-postgresql-1/charm/venv/lib/python3.10/site-packages/urllib3/connectionpool.py", line 841, in urlopen
    retries = retries.increment(
  File "/var/lib/juju/agents/unit-postgresql-1/charm/venv/lib/python3.10/site-packages/urllib3/util/retry.py", line 519, in increment
    raise MaxRetryError(_pool, url, reason) from reason  # type: ignore[arg-type]
urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='10.96.199.166', port=8008): Max retries exceeded with url: /cluster (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fd9b5e0ee90>: Failed to establish a new connection: [Errno 111] Connection refused'))

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-postgresql-1/charm/src/cluster.py", line 568, in are_replicas_up
    response = requests.get(
  File "/var/lib/juju/agents/unit-postgresql-1/charm/venv/lib/python3.10/site-packages/requests/api.py", line 73, in get
    return request("get", url, params=params, **kwargs)
  File "/var/lib/juju/agents/unit-postgresql-1/charm/venv/lib/python3.10/site-packages/requests/api.py", line 59, in request
    return session.request(method=method, url=url, **kwargs)
  File "/var/lib/juju/agents/unit-postgresql-1/charm/venv/lib/python3.10/site-packages/requests/sessions.py", line 589, in request
    resp = self.send(prep, **send_kwargs)
  File "/var/lib/juju/agents/unit-postgresql-1/charm/venv/lib/python3.10/site-packages/requests/sessions.py", line 703, in send
    r = adapter.send(request, **kwargs)
  File "/var/lib/juju/agents/unit-postgresql-1/charm/venv/lib/python3.10/site-packages/requests/adapters.py", line 700, in send
    raise ConnectionError(e, request=request)
requests.exceptions.ConnectionError: HTTPConnectionPool(host='10.96.199.166', port=8008): Max retries exceeded with url: /cluster (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fd9b5e0ee90>: Failed to establish a new connection: [Errno 111] Connection refused'))
machine-0: 01:45:55 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-0: 01:45:55 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-postgresql-0: 01:45:55 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
machine-2: 01:53:10 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-2: 01:53:10 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-haproxy-0: 01:53:10 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
machine-3: 01:53:28 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-3: 01:53:28 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-landscape-server-0: 01:53:29 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
machine-5: 01:53:48 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-5: 01:53:48 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-rabbitmq-server-0: 01:53:48 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
machine-4: 01:54:03 ERROR juju.worker.dependency "lxd-container-provisioner" manifold worker returned unexpected error: container types not yet available
machine-4: 01:54:03 ERROR juju.worker.dependency "kvm-container-provisioner" manifold worker returned unexpected error: container types not yet available
unit-landscape-server-1: 01:54:03 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
unit-pgbouncer-0: 01:56:02 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
unit-pgbouncer-1: 01:56:37 ERROR juju.worker.meterstatus error running "meter-status-changed": charm missing from disk
unit-landscape-server-0: 02:03:24 ERROR unit.landscape-server/0.juju-log Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/storm/exceptions.py", line 165, in wrap_exceptions
    yield
  File "/usr/lib/python3/dist-packages/storm/databases/postgres.py", line 438, in raw_connect
    return self._raw_connect()
  File "/usr/lib/python3/dist-packages/storm/databases/postgres.py", line 419, in _raw_connect
    raw_connection = ConnectionWrapper(psycopg2.connect(self._dsn), self)
  File "/usr/lib/python3/dist-packages/psycopg2/__init__.py", line 122, in connect
    conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
psycopg2.OperationalError: connection to server at "localhost" (127.0.0.1), port 6432 failed: FATAL:  server login has been failing, try again later (server_login_retry)


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/opt/canonical/landscape/bootstrap-account", line 7, in <module>
    canonical.landscape.scripts.bootstrap_account.run()
  File "/opt/canonical/landscape/canonical/landscape/scripts/bootstrap_account.py", line 77, in run
    run_bootstrap_account(sys.argv[1:])
  File "/opt/canonical/landscape/canonical/landscape/scripts/bootstrap_account.py", line 62, in run_bootstrap_account
    bootstrap_account_helper(
  File "/opt/canonical/landscape/canonical/landscape/api/account.py", line 656, in bootstrap_account_helper
    account = get_account_by_name("standalone")
  File "/opt/canonical/landscape/canonical/landscape/model/main/account.py", line 1682, in get_account_by_name
    collection = AccountCollection(get_main_store(), account_names=[name])
  File "/opt/canonical/landscape/canonical/landscape/model/main/store.py", line 19, in get_main_store
    return get_store_by_name("main")
  File "/opt/canonical/landscape/canonical/landscape/model/store.py", line 57, in get_store_by_name
    store = zstorm.get(name)
  File "/usr/lib/python3/dist-packages/storm/zope/zstorm.py", line 179, in get
    return self.create(name, default_uri)
  File "/usr/lib/python3/dist-packages/storm/zope/zstorm.py", line 155, in create
    store = Store(database)
  File "/usr/lib/python3/dist-packages/storm/store.py", line 83, in __init__
    self._connection = database.connect(self._event)
  File "/usr/lib/python3/dist-packages/storm/database.py", line 584, in connect
    return self.connection_factory(self, event)
  File "/usr/lib/python3/dist-packages/storm/database.py", line 270, in __init__
    self._raw_connection = self._database.raw_connect()
  File "/usr/lib/python3/dist-packages/storm/databases/postgres.py", line 437, in raw_connect
    with wrap_exceptions(self):
  File "/usr/lib/python3.10/contextlib.py", line 153, in __exit__
    self.gen.throw(typ, value, traceback)
  File "/usr/lib/python3/dist-packages/storm/exceptions.py", line 184, in wrap_exceptions
    six.raise_from(wrapped.with_traceback(tb), e)
  File "<string>", line 3, in raise_from
  File "/usr/lib/python3/dist-packages/storm/exceptions.py", line 165, in wrap_exceptions
    yield
  File "/usr/lib/python3/dist-packages/storm/databases/postgres.py", line 438, in raw_connect
    return self._raw_connect()
  File "/usr/lib/python3/dist-packages/storm/databases/postgres.py", line 419, in _raw_connect
    raw_connection = ConnectionWrapper(psycopg2.connect(self._dsn), self)
  File "/usr/lib/python3/dist-packages/psycopg2/__init__.py", line 122, in connect
    conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
storm.database.OperationalError: connection to server at "localhost" (127.0.0.1), port 6432 failed: FATAL:  server login has been failing, try again later (server_login_retry)



[32mINFO    [0m pytest_operator.plugin:plugin.py:1025 Forgetting model main...